src/System.Text.Json/src/System/Text/Json/Node/JsonBoolean.cs

```diff
@@ -65,7 +71,15 @@ public sealed class JsonBoolean : JsonNode, IEquatable<JsonBoolean>
         ///   <see langword="true"/> if values of instances match,
         ///   <see langword="false"/> otherwise.
         /// </returns>
-        public static bool operator ==(JsonBoolean left, JsonBoolean right) => left?.Value == right?.Value;
+        public static bool operator ==(JsonBoolean left, JsonBoolean right)
+        {
+            if (right is null)
```

Comment:
It would be good to retain the comment which explains why the code is written this way and link to the coreclr issue.